### PR TITLE
docs(font): add a typesetting section

### PR DESCRIPTION
**Pull request by kisvegabor**
Related to #7190

cc @bryghtlabs-richard @vwheeler63 

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by vwheeler63 — APPROVED** [review]
Nice work.  What happened to "Subpixel rendering"?  Did it go away?

**Comment by kisvegabor** [comment]
> What happened to "Subpixel rendering"? Did it go away?

It's removed from v9.

**Comment by vwheeler63** [comment]
> > What happened to "Subpixel rendering"? Did it go away?
> 
> Yes, it's removed from v9.

Oh wow!  Did anyone complain?

**Comment by bryghtlabs-richard** [comment]
Thanks! Looks good to me.

**Comment by vwheeler63** [comment]
> Yes, it's removed from v9.

Also, does that mean there is no more reason to have 2bpp and 4bpp fonts?

File: docs/details/main-components/font.rst [hunk 1]
```diff
@@ -45,6 +45,31 @@ To test it try
...
+The standard Latin-based languages (e.g., English, Spanish, German)
+and East Asian languages such as Chinese, Japanese, and Korean (CJK)
+are relatively straightforward, as their characters are simply
+written from left to right.
+
+Languages like Arabic, Persian, and Hebrew, which use Right-to-Left
+(RTL) or mixed writing directions, are also supported in LVGL.
+Learn more :ref:`here <bidi>`.
+
+For characters such as ``é`` ``ü``, ``ő``, ``Á``, and ``Ű``, it is recommended
```
Comment (vwheeler63) [inline docs/details/main-components/font.rst]:
```suggestion

.. |Aacute| unicode:: U+000C1 .. LATIN CAPITAL LETTER A WITH ACUTE
.. |eacute| unicode:: U+000E9 .. LATIN SMALL LETTER E WITH ACUTE
.. |otilde| unicode:: U+000F5 .. LATIN SMALL LETTER O WITH TILDE
.. |Utilde| unicode:: U+00168 .. LATIN CAPITAL LETTER U WITH TILDE
.. |uuml|   unicode:: U+000FC .. LATIN SMALL LETTER U WITH DIAERESIS

For characters such as '|eacute|', '|uuml|', '|otilde|', '|Aacute|', and '|Utilde|', it is recommended
```

**Review by vwheeler63 — CHANGES_REQUESTED** [review]

File: docs/details/main-components/font.rst (hunk 1 above)
Comment (vwheeler63) [reply docs/details/main-components/font.rst]: FYI, Sphinx doesn't know what to do with characters outside the ASCII range but does well with the "replacement" directives as above.  Note the in-line literal back-quotes hides the replacement string from the `reStructuredText` parser unfortunately, so the letters have to be enclosed in single- or double-quotes or some other character that doesn't switch the parser into a "different mode".

**Comment by bryghtlabs-richard** [comment]
> > Yes, it's removed from v9.
> 
> Also, does that mean there is no more reason to have 2bpp and 4bpp fonts?

I think subpixel rendering is something different - using knowledge of subpixel(RGB) layout to influence font rendering or baking. We use 2bpp and 4bpp fonts often to save space and improve draw time, so I hope they aren't going away.

File: docs/details/main-components/font.rst (hunk 1 above)
Comment (kisvegabor) [reply docs/details/main-components/font.rst]: Wao, thank you!

**Comment by kisvegabor** [comment]
> We use 2bpp and 4bpp fonts often to save space and improve draw time, so I hope they aren't going away.

They won't go away :slightly_smiling_face: sub pixel rendering was not really used. 2 and 4 bit fonts are used every day!

**Review by vwheeler63 — APPROVED** [review]

**Review by XuNeo — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**